src/ray/core_worker/core_worker.cc

```diff
@@ -2596,8 +2596,15 @@ void CoreWorker::HandleDeleteSpilledObjects(
 
 void CoreWorker::HandleExit(const rpc::ExitRequest &request, rpc::ExitReply *reply,
                             rpc::SendReplyCallback send_reply_callback) {
+  bool own_objects = reference_counter_->OwnObjects();
+  // Fail the request if it owns any object.
+  if (own_objects) {
+    reply->set_success(false);
+  } else {
+    reply->set_success(true);
+    Exit(/*intentional=*/true);
+  }
   send_reply_callback(Status::OK(), nullptr, nullptr);
```

Comment:
Is it necessary to send the reply only after `Exit()` is invoked? If so, maybe add a comment about the reason?

Reply:
It is always sent? Or did I misunderstand your question? What do you mean by it is necessary to send reply "only after"?

Reply:
I mean you changed the order of invoking `Exit` and `send_reply_callback`. Is there a particular reason?

Reply:
oh I see. No reasons. Just to make code look cleaner haha. (because otherwise I need to have another if after sending a reply)